cargo/cargo_manifest.bzl

```diff
@@ -0,0 +1,167 @@
+"""Aspect that generates Cargo.toml files from BUILD files, which is useful
+for interacting with vanilla Rust tooling.
+"""
+
+load("@io_bazel_rules_rust//rust:private/rustc.bzl", "CrateInfo")
+load("@io_bazel_rules_rust//rust:private/utils.bzl", "relativize")
+
+CargoManifestInfo = provider(
+    doc = "A provider contianing information about a Crate's cargo metadata.",
+    fields = {
+        "toml": "The current crate's Cargo.toml file.",
+        "deps": "The Cargo.toml files the current crate depends on.",
+    },
+)
+
+def _output_dir(ctx):
+    """Returns the output directory the aspect should write into.
+
+    Args:
+        ctx (ctx): The current aspect's context object
+
+    Returns:
+        string: The output directory name for the current aspect
+    """
+    return "cargo_manifest.aspect/{}".format(ctx.rule.attr.name)
+
+def _is_external_crate(target):
+    """Returns whether or not the target is an external target.
+
+    eg: `@bazel_skylib//:lib` is an external target where as `@io_bazel_rules_rust//:lib` or `//:lib` are not.
+
+    Args:
+        target (Target): The target the aspect is being applied to.
+
+    Returns:
+        bool: True if the target is an external target
+    """
+    return target.label.workspace_root.startswith("external")
+
+def _clone_external_crate_sources(ctx, target):
```

Comment:
This should ideally be unnecessary but when resolving the generaged Cargo manifest for external crates, the sources could not be found. This copies them so `cbindgen` (and ideally other rules) can locate those files.